packages/core/src/server/hmrFallback.ts

```diff
@@ -0,0 +1,75 @@
+import { promises as dns } from 'node:dns';
+import { DevConfig, ServerConfig } from 'src/types/config';
+
+type Hostname = {
+  host?: string;
+  name: string;
+};
+
+const wildcardHosts = new Set([
+  '0.0.0.0',
+  '::',
+  '0000:0000:0000:0000:0000:0000:0000:0000',
+]);
+
+export async function resolveHostname(
+  optionsHost: string | boolean | undefined,
```

Comment:
Rsbuild's `server.host` only supports `string` type, maybe we can remove `boolean` type here.